wasted-app/RestaurantRegistrationControl.cs

```diff
@@ -40,45 +39,53 @@ private bool checkIfTextBoxesAreFull()
             else return false;
         }
 
-        private void resetTextBoxes()
+        private void ResetTextBoxes()
         {
             restaurantNameTextBox.Text = "";
             latitudeTextBox.Text = "";
             longitudeTextBox.Text = "";
             mailTextBox.Text = "";
             passwordTextBox.Text = "";
             repeatPasswordTextBox.Text = "";
-            textBoxLostFocus("Restaurant name", restaurantNameTextBox);
-            textBoxLostFocus("Latitude", latitudeTextBox);
-            textBoxLostFocus("Longitude", longitudeTextBox);
-            textBoxLostFocus("Mail", mailTextBox);
-            textBoxLostFocus("Password", passwordTextBox);
-            textBoxLostFocus("Repeat Password", repeatPasswordTextBox);
+            TextBoxLostFocus("Restaurant name", restaurantNameTextBox);
+            TextBoxLostFocus("Latitude", latitudeTextBox);
+            TextBoxLostFocus("Longitude", longitudeTextBox);
+            TextBoxLostFocus("Mail", mailTextBox);
+            TextBoxLostFocus("Password", passwordTextBox);
+            TextBoxLostFocus("Repeat Password", repeatPasswordTextBox);
         }
 
         private void registerButton_Click(object sender, EventArgs e)
         {
-            if(passwordTextBox.Text == repeatPasswordTextBox.Text && checkIfTextBoxesAreFull())
+            if(passwordTextBox.Text == repeatPasswordTextBox.Text && CheckIfTextBoxesAreFull())
             {
+                ServicesController controller = ServicesController.Instance;
                 var mail = mailTextBox.Text;
                 var password = passwordTextBox.Text;
-                String credentialError = getValidationError(mail, password);
-                if (credentialError == "")
+                var restaurant = controller.RestaurantService.GetByMail(new Mail(mail));
+                if (restaurant == null)
```

Comment:
I believe that we should use the method "Register(Credentials, Restaurant) that returns a bool" instead of this validation here. This validation will be implemented on the register method. 
We can also throw an exception in the method, so that you can catch it here and just display the error message on the screen.